tensorboard/manager.py

```diff
@@ -300,3 +302,129 @@ def get_all():
     else:
       results.append(info)
   return results
+
+
+# Indicates that a call to `start` was compatible with an existing
+# TensorBoard process, which can be reused according to the provided
+# info.
+StartReused = collections.namedtuple("StartReused", ("info",))
+
+# Indicates that a call to `start` successfully launched a new
+# TensorBoard process, which is available with the provided info.
+StartLaunched = collections.namedtuple("StartLaunched", ("info",))
+
+# Indicates that a call to `start` tried to launch a new TensorBoard
+# instance, but the subprocess exited with the given exit code and
+# output streams. (If the contents of the output streams are no longer
+# available---e.g., because the user has emptied /tmp/---then the
+# corresponding values will be `None`.)
+StartFailed = collections.namedtuple(
+    "StartFailed",
+    (
+        "exit_code",  # int, as `Popen.returncode` (negative for signal)
+        "stdout",  # str, or `None` if the stream could not be read
+        "stderr",  # str, or `None` if the stream could not be read
+    ),
+)
+
+# Indicates that a call to `start` launched a TensorBoard process, but
+# that process neither exited nor wrote its info file within the allowed
+# timeout period. The process may still be running under the included
+# PID.
+StartTimedOut = collections.namedtuple("StartTimedOut", ("pid",))
+
+
+def start(arguments, timeout=datetime.timedelta(seconds=10)):
+  """Start a new TensorBoard instance, or reuse a compatible one.
+
+  If the cache key determined by the provided arguments and the current
+  working directory (see `cache_key`) matches the cache key of a running
+  TensorBoard process (see `get_all`), that process will be reused.
+
+  Otherwise, a new TensorBoard process will be spawned with the provided
+  arguments, using the `tensorboard` binary from the system path.
+
+  Args:
+    arguments: List of strings to be passed as arguments to
+      `tensorboard`. (If you have a raw command-line string, see
+      `shlex.split`.)
+    timeout: `datetime.timedelta` object describing how long to wait
+      after launching the subprocess before giving up on it. If the
```

Comment:
This comment is not that easy to understand. What does "giving up" mean? Who will leave it running in the background and when?

Reply:
Thanks; will clarify.